### PR TITLE
feat: increase rfqm gas estimates for certain tokens

### PR DESCRIPTION
**Pull request by phil-ociraptor**
# Description

# Testing Instructions

# Checklist

-   [ ] Update [documentation](https://github.com/0xProject/website/blob/development/mdx/api/index.mdx) as needed. **Website Documentation PR:**
-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.
-   [ ] Test changes on the staging environment with [Matcha API Staging](https://api-staging.matcha.xyz).

    -   [ ] SRA/Limit orders
    -   [ ] Swap endpoints
    -   [ ] Meta transaction endpoints
    -   [ ] Depth charts

    For more information see `0x API Matcha smoke test runbook` in Quip.

### CODE REVIEW

File: src/utils/rfqm_gas_estimate_utils.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+const TOKEN_GAS_PREMIUM: Record<string, number> = {
+    '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48': 20e3, // USDC
+    '0xdac17f958d2ee523a2206206994597c13d831ec7': 20e3, // USDT
+};
+
+/**
+ * Prepares the gas estimate for an RFQM trade
+ */
+export function calculateGasEstimate(makerToken: string, takerToken: string): number {
+    const makerTokenPremium: number = TOKEN_GAS_PREMIUM[makerToken] || 0;
```
Comment (PirosB3) [inline src/utils/rfqm_gas_estimate_utils.ts:12]: Can we force a `toLowerCase` here to be safe?

File: src/utils/rfqm_gas_estimate_utils.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { RFQM_TX_GAS_ESTIMATE } from '../constants';
+
+const TOKEN_GAS_PREMIUM: Record<string, number> = {
+    '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48': 20e3, // USDC
+    '0xdac17f958d2ee523a2206206994597c13d831ec7': 20e3, // USDT
```
Comment (PirosB3) [inline src/utils/rfqm_gas_estimate_utils.ts:5]: What about WBTC? Does that have an overhead?

**Review by PirosB3 — APPROVED** [review]
Added one small comment

File: test/utils/rfqm_gas_estimate_utils_test.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        it('should add a token premium for taker token', () => {
+            const makerToken = '';
+            const takerToken = '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48';
+
+            const result = calculateGasEstimate(makerToken, takerToken);
+
+            expect(result).to.be.greaterThan(RFQM_TX_GAS_ESTIMATE);
+        });
+
+        it('should token premiums for maker and taker token should be additive', () => {
```
Comment (rhinodavid) [inline test/utils/rfqm_gas_estimate_utils_test.ts:39]:
```suggestion
        it('token premiums for maker and taker token should be additive', () => {
```

File: test/utils/rfqm_gas_estimate_utils_test.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
+// tslint:disable:custom-no-magic-numbers
+// tslint:disable:no-empty
```
Comment (rhinodavid) [inline test/utils/rfqm_gas_estimate_utils_test.ts:2]:
```suggestion
```

File: src/utils/rfqm_gas_estimate_utils.ts (hunk 2 above)
Comment (phil-ociraptor) [reply src/utils/rfqm_gas_estimate_utils.ts:5]: not from what I have seen, but if the numbers start to show that it elevates gas costs, we can add an entry

File: src/utils/rfqm_gas_estimate_utils.ts (hunk 1 above)
Comment (phil-ociraptor) [reply src/utils/rfqm_gas_estimate_utils.ts:12]: good call

**Pull request merged**